cf-java-logging-support-core/src/main/java/com/sap/hcp/cf/logging/common/converter/DefaultPropertiesConverter.java

```diff
@@ -46,7 +52,14 @@ public void convert(StringBuilder appendTo, Map<String, String> eventProperties)
                 ObjectComposer<JSONComposer<String>> oc = JSON.std.composeString().startObject();
                 for (Entry<String, String> p: properties.entrySet()) {
                     if (!exclusions.contains(p.getKey())) {
-                        oc.put(p.getKey(), p.getValue());
+                        if (sendDefaultValues) {
+                            oc.put(p.getKey(), p.getValue());
+                        } else {
+                            String defaultValue = getDefaultValue(p.getKey());
+                            if (!defaultValue.equals(p.getValue())) {
```

Comment:
this will only ever omit fields, for which a default is known right?
as the value will never be equal to the unknown Defaults value, if none is present.

Reply:
The nice thing is, that it falls back to `Defaults.UNKNOWN`, which is "-". So this should be a reasonable approach even for customer provided fields.